src/iuse.cpp

```diff
@@ -4661,11 +4662,52 @@ cata::optional<int> iuse::vortex( Character *p, item *it, bool, const tripoint &
 
 cata::optional<int> iuse::dog_whistle( Character *p, item *it, bool, const tripoint & )
 {
+    if( !p->is_avatar() ) {
+        return cata::nullopt;
+    }
     if( p->is_underwater() ) {
         p->add_msg_if_player( m_info, _( "You can't do that while underwater." ) );
         return cata::nullopt;
     }
     p->add_msg_if_player( _( "You blow your dog whistle." ) );
+
+    std::array<std::string, 4> messages_friendly_or_neutral = {
+        _( "What is this unbearable sound!?" ),
+        _( "STOP.  MY EARS" ),
+        _( "I'm not a dog…" ),
+        _( "Would you kindly not do that?" )
+    };
+
+    std::array<std::string, 5> messages_hostile = {
+        _( "I WILL MURDER YOU" ),
+        _( "I'LL SHOVE THAT WHISTLE DOWN YOUR THROAT" ),
+        _( "You're seriously pissing me off…" ),
+        _( "I'm not a dog…" ),
+        _( "What is this unbearable sound!?" ),
+    };
+
+    // Can the Character hear the dog whistle?
+    auto hearing_check = [p]( const Character & who ) -> bool {
+        return !who.is_deaf() && p->sees( who ) &&
+        who.has_trait( STATIC( trait_id( "THRESH_LUPINE" ) ) );
+    };
+
+    for( const npc &subject : g->all_npcs() ) {
+        if( !( one_in( 3 ) && hearing_check( subject ) ) ) {
+            continue;
+        }
+
+        if( p->attitude_to( subject ) == Creature::Attitude::HOSTILE ) {
+            subject.say( random_entry( messages_hostile ) );
+        } else {
+            subject.say( random_entry( messages_friendly_or_neutral ) );
+        }
+    }
+
+    if( hearing_check( *p ) && one_in( 3 ) ) {
+        p->add_msg_if_player( m_info, _( "You hate this loud sound." ) );
```

Comment:
Would it make sense to add a morale malus here?

Reply:
I decided on it not having any impact on the game, just a funny message for the player, but a morale effect ( or making NPCs angry ) could be added. What value and duration would you suggest?

Reply:
I was thinking something fairly small and maybe a ten minute length.  But I think what you've done is reasonable.